code/datums/elements/hud_maps.dm

```diff
@@ -0,0 +1,104 @@
+
+/**
+ * Bespoke Element for mobs to access when they want to view a HUD of the map
+ * Keep it bespoke so we have to update as few objects as possible
+ */
+/datum/element/hud_map
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	var/obj/screen/minimap/map
+	var/mapz = 0
+	var/filter = NONE
+
+/datum/element/hud_map/Attach(mob/target, ztarget, filtertargets)
+	if(!ismob(target))
+		return ELEMENT_INCOMPATIBLE
+	. = ..()
+	RegisterSignal(target, COMSIG_MOVABLE_Z_CHANGED, .proc/change_map)
+	if(!SSminimaps.minimaps_by_z["[ztarget]"].hud_image)
+		return
+	if(!map)
+		mapz = ztarget
+		filter = filtertargets
+		map = new(null, ztarget, filtertargets)
+	RegisterSignal(target, COMSIG_MOB_HUDMAP_TOGGLED, .proc/display_map_to)
+
+///Displays the HUD display to the inputted mob source
+/datum/element/hud_map/proc/display_map_to(mob/source)
+	SIGNAL_HANDLER
+	source.client.screen += map
+	UnregisterSignal(source, COMSIG_MOB_HUDMAP_TOGGLED)
+	RegisterSignal(source, COMSIG_MOB_HUDMAP_TOGGLED, .proc/undisplay_map)
+
+///Undisplays the HUD display to the inputted mob source
+/datum/element/hud_map/proc/undisplay_map(mob/source)
+	SIGNAL_HANDLER
+	source.client.screen -= map
+	UnregisterSignal(source, COMSIG_MOB_HUDMAP_TOGGLED)
+	RegisterSignal(source, COMSIG_MOB_HUDMAP_TOGGLED, .proc/display_map_to)
```

Comment:
Why not having a toggled proc as a unique signal handler with a bool on the hud map? I know registering signals is cheap, but what you are doing seems unnecessary

Reply:
As said in discor, its an element